request-strategy/order.go

```diff
@@ -67,22 +67,24 @@ func GetRequestablePieces(
 
 		ih := _i.key.InfoHash
 		t := input.Torrent(ih)
+		if t.IgnorePiece(_i.key.Index) {
```

Comment:
I don't believe this can be moved up like this.

Reply:
hm, could you help me understand why not?
Previous code:
- get torrent
- check if storage is full (quite unlikely)
- chef if piece should be ignored (quite likely)

Updated:
- get torrent
- chef if piece should be ignored (quite likely)
- check if storage is full (quite unlikely)
I don't think there is drastic difference, but feels like It can save a couple of cpu cycles in cases with millions of pieces, I don't mind discarding this change.

Reply:
Yes, let me try:

If a piece is ignored (not wanted, which doesn't preclude already having it), it may still have a priority that makes it desirable to occupy storage space. So we want to the piece against the available storage capacity, even if we don't need to request it (because we already have it most likely).

Additionally, the change you made move the unverified bytes check before the total storage check. I don't think this has any impact on behaviour (assuming it returns false like the storage capacity check). I wonder which is faster.

Thanks for your efforts to look into this.